Fix to GenerateChallenge

DenisKarch opened the pull request:
crypto/x509 now enforces that rsa public keys must have NULL parameters.
The old hack no longer solves the issue and instead will silently fail when
parsing the key, ultimately causing a null pointer dereference at
(pubkey := cert.PublicKey.(*rsa.PublicKey)).
Currently working with crypto/x509 to add support for RSAES-OAEP keys
https://github.com/golang/go/issues/30416

ericchiang commented:
Wait, this change was made in certificate transparency, not in crypto/x509 https://github.com/google/certificate-transparency-go/pull/470

To use this in go-tspi, we need to use that package right?

DenisKarch committed:
Fix to GenerateChallenge

crypto/x509 now enforces that rsa public keys must have NULL parameters.
The old fix no longer solves the issue and instead will silently fail when
parsing the key, ultimately causing a null pointer dereference at
(pubkey := cert.PublicKey.(*rsa.PublicKey)).
Currently working with crypto/x509 to add support for RSAES-OAEP keys
golang/go#30416

certificate-transparency-go have accepted a temporary fix to /x509
google/certificate-transparency-go#470
so we will be building against them for the time being.

DenisKarch commented:
Yep, changed the import to build against google/certificate-transparency-go/x509 instead
